### PR TITLE
[FIX] registry, loading: re-init inherited SQL views

### LINKED ISSUE #23528: [11.0] Uninstalling sale_stock is breaking Sale module's feature
Impacted versions: 11.0

Steps to reproduce: Uninstall sale_stock module

Current behavior: Go to Sales --> Reporting -->Sales

Expected behavior: Report Should be there.

Video/Screenshot link (optional):

### PR DESCRIPTION
**Pull request by Elkasitu**
Before this commit:

* Install any module that creates a SQL view and another that extends
this view.
    e.g.: sale and pos_sale for `report.all.channels.sales`.

* Uninstall the module that extended the SQL view.
    e.g.: uninstall pos_sale.

* Try to access the view from the web client -> Traceback, table not
found.

This happens because when reloading the registry, postgres drops the sql
view and it must be re-initialized.

After this commit:

We solve this issue by calculating all missing tables/views during the
uninstallation process, and re-initializing all of them right before
reloading the registry.

Fixes #23528, #23529, #23530

TODO: Remove hacks in uninstall_hooks...

### CODE REVIEW

**Comment by Elkasitu** [comment]
@dbo-odoo probably of interest to you since I remember you complaining about SQL views and uninstall issues, I guess this was it :)

File: addons/sale/__init__.py [hunk 1]
```diff
@@ -16,8 +16,6 @@ def update_dashboard_graph_model(dbname):
         db_registry = odoo.modules.registry.Registry.new(dbname)
         with api.Environment.manage(), db_registry.cursor() as cr:
             env = api.Environment(cr, SUPERUSER_ID, {})
-            if 'account.invoice.report' in env:
-                env['account.invoice.report'].init()
```
Comment (dbo-odoo) [inline addons/sale/__init__.py]: mais lol

**Comment by dbo-odoo** [comment]
@Elkasitu 
excellent news ideed

**Comment by rvalyi** [comment]
sure this doesn't affect v9 or v10 too? Just asking...

**Comment by Elkasitu** [comment]
@rvalyi It does, but it will be backported only if needed

**Comment by Elkasitu** [comment]
@dbo-odoo Do you reckon it'd be best to merge this into v9 directly or should I merge into v11 and then backport as needed?

**Pull request merged**